### PR TITLE
Some fixes

### LINKED ISSUE #237: Check rate limiter
* nechceme abychom zbytecne blokovali lidi na stejne IP adrese, je treba zkontrolovat, zdali vubec v AWS zname tu spravnou IP a neblokujeme toho zbytecne moc

### LINKED ISSUE #243: Time around interval
Dnes zobrazime, ze se pacient ma dostavit 15:14-15:17. To vypada dementne, tak bych k zacatku a konci vzdy pridal 10 minut, aby to byl pulhodinovy interval. Idealne toto tedy 15:04-15:27, pripadne kdyby to bylo tezke, tak proste jak to pujde -> aby pacient dostal vetsi interval.

### LINKED ISSUE #248: Bad request
271d21e6-ee69-4fc9-b577-13828bc32f33
0baf3b19-930c-44c9-aa1f-dcc5870ce176
eece198b-26dd-4302-b484-900a5d238152
5247f266-3091-42f8-a34c-6baff848ce96
 je bad request bez dalsich detailu. Tezko rict, jestli s tim muzeme neco delat.

### PR DESCRIPTION
**Pull request by LukasForst**
* better logging when throwing an error -> closes #248 
* round time slots -> closes #243 
* reasonable default rate limit -> closes #237 
* some refactoring and a lot of tests (routes >= 90% coverage, whole app >= 60%)
* added tasks for students -> see #262, #259, #258, #256, #255 
* ISIN validation is disabled by default -> @marazt it is necessary to update production configuration and add env `ENABLE_ISIN_PATIENT_VALIDATION=true`

### CODE REVIEW

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt [hunk 1]
```diff
@@ -29,74 +29,76 @@ private val logger = createLogger("ExceptionHandler")
...
  * Registers exception handling.
  */
+@Suppress("LongMethod") // it's fine as this is part of KTor
 fun Application.installExceptionHandling() {
     install(StatusPages) {
-        // TODO define correct logging level policy
-
         exception<InsufficientRightsException> {
-            logger.warn {
+            logger.warn(it) {
```
[the remark this answers is not in the document]
Comment (tomaskourim) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:36]: Nicmene bys mohl mozna...

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt [hunk 2]
```diff
@@ -29,74 +29,76 @@ private val logger = createLogger("ExceptionHandler")
...
+            call.errorResponse(HttpStatusCode.NotFound, it.message)
         }
 
         jsonExceptions()
 
-        exception<IsinValidationException> { cause ->
-            logger.debug { cause.message }
-            call.errorResponse(HttpStatusCode.NotAcceptable, "Not acceptable: ${cause.message}.")
+        exception<IsinValidationException> {
+            logger.info(it) { "ISIN validation failed - $it." }
```
Comment (tomaskourim) [inline backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:72]: proc jenom info?

**Review by tomaskourim — CHANGES_REQUESTED** [review]
Rychle jsem to prolitl. Vypada to ok. Ale urcite s mergovanim pockejme nez bude nova ENV promenna nastavena. A take by to podle me chtel videl @tomaspavlin, protoze to upravuje i jeho kod.

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukasForst) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:36]: tohle se pouzije, protoze se to propise do finalniho jsonu jako `"exception": {...}` tzn v AWS uvidime celej stacktrace - to mi prijde uzitecne

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt (hunk 2 above)
Comment (LukasForst) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:72]: hele tot otazka, nevim jak casto se bude stavat, ze nam nekdo zada cloveka, co neexistuje, ale asi muzu hodit warn, snad to nebude tolik plasit

**Comment by tomaskourim** [comment]
Hele za mě je to ok jenom nesmíme zapomenout na ten env na produkci

**Comment by LukasForst** [comment]
image
env je nastavena -> related PR https://github.com/mild-blue/project-configuration/pull/43

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomaskourim) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:36]: Jakoze bez toho `it` se nevypise stacktrace, jo? Potom to samozrejme chceme

**Review by tomaskourim — APPROVED** [review]
velky PR, rychle zkontrolovany a vypada gut

File: backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukasForst) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/error/ExceptionHandling.kt:36]: presne tak

**Pull request merged**